### PR TITLE
Rearrange npfg use path input

### PR DESCRIPTION
**Pull request by KonradRudin**
### Solved Problem
Rearranged the interaction between the FwPosController and npfg. the guidance logic should not be concerned with the waypoint triplets directly. The FwPosController converts the triplets into a path setpoint which is fed into the npfg navigation logic. This is part of separating the npfg into smaller controllers with cleaner API.

### Solution
- Move the navigateXX function into the FwPosController. They are responsible to convert the setpoint triplets into a path setpoint.
- Update the npfg to only rely on one single navigatePath input function, which is used to navigate to the desired path.

### Test coverage
- Tested in SImulation using the standard_vtol gazebo simulation.

### Context
Related links, screenshot before/after, video

### CODE REVIEW

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 1]
```diff
@@ -1279,10 +1280,17 @@ FixedwingPositionControl::control_auto_loiter(const float control_interval, cons
...
+	Vector2f vehicle_to_loiter_center{curr_wp_local - curr_pos_local};
+
+	static constexpr float LOITER_REACHED_DISTANCE =
+		10.f; // Expected distnce to loiter radius from which we assume to have it reached. // TODO make a parameter out of it.
+
+	const bool close_to_circle = vehicle_to_loiter_center.norm() < loiter_radius + LOITER_REACHED_DISTANCE;
 
 	if (pos_sp_next.type == position_setpoint_s::SETPOINT_TYPE_LAND && _position_setpoint_next_valid
-	    && in_circle_mode && _param_fw_lnd_earlycfg.get()) {
+	    && close_to_circle && _param_fw_lnd_earlycfg.get()) {
```
[the remark this answers is not in the document]
Comment (tstastny) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:1293]: maybe while we're at it, could also move
```cpp
} else {
	_att_sp.apply_flaps = vehicle_attitude_setpoint_s::FLAPS_OFF;
	_att_sp.apply_spoilers = vehicle_attitude_setpoint_s::SPOILERS_OFF;
}
```
out of the else condition and just set it in the same place as the original `airspeed_sp` initialization. Then it just gets overwritten by the if condition. slightly less messy? (also makes it easier later to see that we anyway need to extract this landing logic out of this orbit method)

**Review by tstastny — CHANGES_REQUESTED** [review]
Thanks for getting this rolling @KonradRudin !

@dagar just heads up - this is first steps in a larger fixed-wing control refactor. end game is split up "path controller" "course controller" and "heading controller". no exotic setpoints, just standard vehicle states.

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfuhrer) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:1293]: Or we get https://github.com/PX4/PX4-Autopilot/pull/20815 in finally and not have to anymore deal with flaps/spoiler setpoints in the attitude setpoint?

[the remark this answers is not in the document]
Comment (KonradRudin) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:1293]: @sfuhrer even with your PR, the logic to set them should still be the same, it only sets them in another uorb topic, right? So i can still change the proposed changes from tstastny?

**Review by tstastny — DISMISSED** [review]
Still some checks failing - but otherwise gtg from my side.

**Comment by KonradRudin** [comment]
@tstastny can you approve again? I just did a rebase on newest master and cleanup of the commits. The clang tidy error was a bug, rerunning it fixed it. The other errors are flash errors which are already present in main.

**Review by tstastny — APPROVED** [review]

**Pull request merged**